request_information.go

```diff
@@ -1,566 +1,579 @@
+			substitutions[key] = request.normalizeParameters(reflect.ValueOf(value), value, false)
+		}
+		for key, value := range request.QueryParameters {
+			substitutions[key] = value
+		}
+		for key, value := range request.QueryParametersAny {
+			substitutions[key] = value
+		}
+		url, err := stduritemplate.Expand(request.UrlTemplate, substitutions)
+		if err != nil {
+			return nil, err
+		}
+		uri, err := u.Parse(url)
+		return uri, err
+	}
+}
+
+// SetUri updates the URI for the request from a raw URL.
+func (request *RequestInformation) SetUri(url u.URL) {
+	request.uri = &url
+	for k := range request.PathParameters {
+		delete(request.PathParameters, k)
+	}
+	for k := range request.QueryParameters {
+		delete(request.QueryParameters, k)
+	}
+	for k := range request.QueryParametersAny {
+		delete(request.QueryParametersAny, k)
+	}
+}
+
+// AddRequestOptions adds an option to the request to be read by the middleware infrastructure.
+func (request *RequestInformation) AddRequestOptions(options []RequestOption) {
+	if options == nil {
+		return
+	}
+	if request.options == nil {
+		request.options = make(map[string]RequestOption, len(options))
+	}
+	for _, option := range options {
+		request.options[option.GetKey().Key] = option
+	}
+}
+
+// GetRequestOptions returns the options for this request. Options are unique by type. If an option of the same type is added twice, the last one wins.
+func (request *RequestInformation) GetRequestOptions() []RequestOption {
+	if request.options == nil {
+		return []RequestOption{}
+	}
+	result := make([]RequestOption, len(request.options))
+	idx := 0
+	for _, option := range request.options {
+		result[idx] = option
+		idx++
+	}
+	return result
+}
+
+const contentTypeHeader = "Content-Type"
+const binaryContentType = "application/octet-steam"
+
+// SetStreamContent sets the request body to a binary stream.
+// Deprecated: Use SetStreamContentAndContentType instead.
+func (request *RequestInformation) SetStreamContent(content []byte) {
+	request.SetStreamContentAndContentType(content, binaryContentType)
+}
+
+// SetStreamContentAndContentType sets the request body to a binary stream with the specified content type.
+// Deprecated: Use SetContentReaderAndContentType instead.
+func (request *RequestInformation) SetStreamContentAndContentType(content []byte, contentType string) {
+	request.Content = content
+	if request.Headers != nil {
+		request.Headers.Add(contentTypeHeader, contentType)
+	}
+}
+
+// SetContentReaderAndContentType sets the request body to a binary stream with the specified content type.
+func (request *RequestInformation) SetContentReaderAndContentType(reader io.ReadCloser, contentType string) {
+	request.ContentReader = reader
+	if request.Headers != nil {
+		request.Headers.Add(contentTypeHeader, contentType)
+	}
+}
+
+func (request *RequestInformation) setContentAndContentType(writer s.SerializationWriter, contentType string) error {
+	content, err := writer.GetSerializedContent()
```

Comment:
I'd expect anything that's setting content to be updated to set the new field instead so implementers of request adapter can have a consistent experience here.
Alternatively, we could keep the code as-is, but add a vanity getter that'd return the reader if set, or wrap the content in a reader and return it if set.